src/test_l1_dsVideoPort.c

```diff
@@ -5020,20 +4741,20 @@ void test_l1_dsVideoPort_positive_dsSetBackgroundColor(void) {
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
 	dsError_t status;
-	intptr_t handle[NUM_OF_PORTS]={HANDLE_ARRAY_INIT};
+	intptr_t handle[gDSvideoPort_NumberOfPorts];
 
 	// Step 01: Initialize video port system
 	status = dsVideoPortInit();
 	UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 	// Step 02: Get the port handle for supported video ports
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &handle[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 03: Set background color for each port by looping through dsVideoBackgroundColor_t enum
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		for(dsVideoBackgroundColor_t color = dsVIDEO_BGCOLOR_BLUE; color < dsVIDEO_BGCOLOR_MAX ; color++){
 			status = dsSetBackgroundColor(handle[i], color);
```

Comment:
check this implement as per header 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsVideoPort.h#L936
please check other API's also

Reply:
done